applications/backend/src/index.ts

```diff
@@ -61,7 +61,17 @@ const io = new Server<CCPSocketEventsMap>(httpServer, {})
 io.on('connection', (socket) => {
 	logger.info(`new connection from ${socket.id}!`)
 
-	socket.on(SETTINGS_UPDATE, (settingsUpdate) => store.dispatch(updateSettings(settingsUpdate)))
+	socket.on(SETTINGS_UPDATE, async (settingsUpdate) => {
+		if (settingsUpdate.twitchChannel !== store.getState().settings.twitchChannel) {
+			await twitchChatHandler.joinNewChannel(
+				settingsUpdate.twitchChannel ?? '',
+				store.getState().settings.twitchChannel
+			)
+		} else {
+			logger.info(`Already connected to channel ${settingsUpdate.twitchChannel}.`)
+		}
```

Comment:
Wouldn't it be better to trigger a channel switch through a state-update. So instead of listening to the socket-event, subscribe to the store (`store.subscribe(() => {...})`) and trigger the channel switch if the `settings.twitchChannel` value has changed. This way the state (`settings.twitchChannel`) could also be changed somewhere else in the app and still have the same behavior of joining the new channel.

Reply:
This is (sounds like) exactly what I have been looking for! This sounds like a good topic for pair-programming for knowledge-sharing about redux :-)